### DISCUSSION TITLE
Installando Mosquitto

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#60] [HUB: Home Assistant]
Ho seguito le istruzioni della pagina di Domus.it per l'installazione del broker MQTT “Mosquitto”.
Arrivato all'esecuzione del comando:

sudo apt-get install mosquitto mosquitto-clients -y

Ottengo:
```
Lettura elenco dei pacchetti... Fatto
Generazione albero delle dipendenze
Lettura informazioni sullo stato... Fatto
mosquitto è già alla versione più recente (2.0.9-0mosquitto1~buster1).
mosquitto-clients è già alla versione più recente (2.0.9-0mosquitto1~buster1).
0 aggiornati, 0 installati, 0 da rimuovere e 6 non aggiornati.

N: Viene ignorato il file "mosquitto-buster.list.1" nella directory "/etc/apt/sources.list.d/" poiché ha un'estensione non valida.

```
Cosa devo fare?

**Comment by snakuzzo** [comment]
Mosquitto è già installato (come puoi vedere dai messaggi). Quindi ok.

Quell'errore che vedi deriva da un tuo errore fatto a monte.
Quando hai seguito il punto **_"Ora installare la lista dei package necessari per il Raspberry"_** hai sicuramente lanciato due volte il comando wget per scaricare il file con il source di mosquitto per apt.
Per non visualizzare più quell'errore ti basta rimuovere il file di troppo che hai scaricato...

`sudo rm /etc/apt/sources.list.d/mosquitto-buster.list.1`

...e provare...

`apt-get update `

Non dovresti più avere errori

Consiglio...se non sei pratico un minimo su Linux, prova prima un tipo di installazione più semplice (HassOS) altrimenti risolvi questo e dopo 10m sarai di nuovo in difficoltà.
Altro consiglio...se proprio vuoi continuare su questa strada...MAI USARE L'OS IN ITALIANO...il sistema operativo va installato con la lingua INGLESE. La lingua italiana per queste cose va dimenticata :) E se hai qualche problema, prendi il messaggio d'errore e fai una ricerca...al 99,99% troverai subito la soluzione al problema ;)

**Reply by Dregi56** [reply]
Risolto, grazie. 
Non ho una giovane età, ho usato anche DOS, Dosshell, ma devo ammettere che il mondo Linux....è veramente altra cosa e sentirsi a proprio agio non è facile e le difficoltà sono dietro l'angolo... ho scelto HA Core perchè nella presentazione era la più consigliata, ma mi stavo chiedendo, se anche da HassSO puoi editare YAML e fare impostazioni via testo come in Core, in cosa Core ti dà di più?  I tuoi consigli sono preziosi, ma ormai trovo problematico ricominciare tutto con una nuova installazione. Nel caso ti chiederò consiglio per un passaggio indolore. Per il momento provo a tener duro. ho visto che piano piano di cose ne ho imparate... e fretta non ne ho :)  
Per quanto riguarda la lingua inglese sul SO, sul PC ho Ubuntu ed è installato in italiano, per cui anche qui avevo scelto l'italiano, anche se effettivamente Domus.it lo sconsigliava.

**Reply by snakuzzo** [reply]
Comunque il nome del sito è "inDomus" :D

A parte questo piccolo dettaglio, se hai tempo e voglia (l'età anagrafica non conta...l'unica cosa che conta è la voglia di imparare e non arrendersi di fronte alle difficoltà) continua con Core.
I vantaggi sono diversi:

1. Impari sicuramente di più
2. Sistema mooooolto più snello
3. Possibilità di utilizzare l'OS su cui sei appoggiato per farci quello che ti pare

Ad ogni modo [QUI](https://indomus.it/focus/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/) se non ricordo male abbiamo fatto un piccolo riassunto

**Reply by Dregi56** [reply]
inDomus certo, il dettaglio non è piccolo... a me dà fastidio quando sbagliano il mio nome :) 
Allora provo a continuare con Core, il fatto di avere il SO "libero" è certamente un vantaggio, mi dà fastidio avere un  computer (Raspberry) dedicato ad una sola cosa, visto che può fare anche altro :)
Comunque, visto che sei molto disponibile con tutti, allora conto sul tuo aiuto.   :)
Tornando a Mosquitto ora, anche grazie al tuo aiuto, il server funziona e sono pronto per avventurarmi verso MQTT, se questo mi permetterà di svincolarmi dai cloud, è la strada giusta. Grazie.
